demos/common/cpp/utils/include/utils/ocv_common.hpp

```diff
@@ -315,3 +315,28 @@ class InputTransform {
     cv::Scalar means;
     cv::Scalar stdScales;
 };
+
+class LazyVideoWriter {
+    cv::VideoWriter writer;
+    unsigned nwritten;
+public:
+    const std::string filenames;
+    const double fps;
+    const unsigned lim;
+
+    LazyVideoWriter(const std::string& filenames, double fps, unsigned lim) :
+        nwritten{1}, filenames{filenames}, fps{fps}, lim{lim} {}
+    void write(cv::InputArray im) {
+        if (writer.isOpened() && (nwritten < lim || 0 == lim)) {
```

Comment:
```suggestion
        if (writer.isOpened() && (nwritten <= lim || 0 == lim)) {
```
Since you are starting with 1

Reply:
It starts with 0. `if (!writer.isOpened() && !filenames.empty())` branch doesn't have `++nwritten`;